conversational-client/src/app/data/conversation.ts

```diff
@@ -7,3 +7,14 @@ export interface ChatMessage {
 // Controls the agent used in a conversation.
 export const AGENT_IDS = ['default', 'fake', 'history_tutor'] as const;
 export type AgentId = (typeof AGENT_IDS)[number];
+
+// Represents a stateful conversational agent.
+export interface Agent {
+  updateState(state: AgentState): AgentState;
+}
+
+// Represents a generic agent state.
+export interface AgentState {
+  messageHistory: ChatMessage[];
```

Comment:
Let's make author an enum as well & follow the naming used by langchain: 
 - ai
 - human

Reply:
can we leave for a follow up PR since it involves server-side changes?